### PR TITLE
feat(packages): Add harbor registryctl, jobservice, and portal

### PR DESCRIPTION
**Pull request by EyeCantCU**
#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

### CODE REVIEW

File: harbor-core.yaml [hunk 1]
```diff
@@ -47,6 +51,74 @@ pipeline:
...
+    description: harbor jobservice
+    pipeline:
+      - uses: go/build
+        with:
+          packages: ./jobservice
+          output: harbor-jobservice
+          modroot: ./src
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}/harbor
+          ln -sf /usr/bin/harbor-jobservice ${{targets.destdir}}/harbor/harbor_jobservice
```
Comment (dlorenc) [inline harbor-core.yaml:65]: This means that the subpackage has to depend on the full package right? Is that intentional?

File: harbor-registry.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/distribution/distribution
+      tag: v${{package.version}}
+      expected-commit: 4772604ae973031ab32dd9805a4bccf61d94909f
+
+  - runs: |
+      # Fetch goharbor's redis patch
+      curl https://raw.githubusercontent.com/goharbor/harbor/main/make/photon/registry/redis.patch -o redis.patch
```
Comment (dlorenc) [inline harbor-registry.yaml:27]: Should we either vendor this patch in this repo or pin it by digest?

Comment (EyeCantCU) [reply harbor-registry.yaml:27]: Vendoring it here is probably way to go. Then we can just drop curl as an env dep

File: harbor-core.yaml (hunk 1 above)
Comment (EyeCantCU) [reply harbor-core.yaml:65]: No, it doesn't have to depend on the full package. There isn't an implicit dependency created here that brings in `harbor-core` so we're good

**Comment by EyeCantCU** [comment]
CI attempts to build core and associated subpackages before the registry causing registryctl's build to fail. I'll split this into another PR

**Comment by EyeCantCU** [comment]
This PR will build after this has been merged: https://github.com/wolfi-dev/os/pull/15496

File: harbor-core.yaml (hunk 1 above)
Comment (EyeCantCU) [reply harbor-core.yaml:65]: OH WAIT. I see what you mean. Fix incoming

**Review by imjasonh — APPROVED** [review]

**Pull request merged**